src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -278,13 +278,14 @@ internal void PublishRpcChannelReadyEvent(RpcEvent initEvent)
                 HandleWorkerError(exc);
                 return;
             }
-            if (_functionRegistrations == null)
+
+            RpcChannelReadyEvent readyEvent = new RpcChannelReadyEvent(_workerId, _workerConfig.Language, this, _initMessage.WorkerVersion, _initMessage.Capabilities);
+            _eventManager.Publish(readyEvent);
+
+            if (_functionRegistrations != null)
             {
-                RpcChannelReadyEvent readyEvent = new RpcChannelReadyEvent(_workerId, _workerConfig.Language, this, _initMessage.WorkerVersion, _initMessage.Capabilities);
-                _eventManager.Publish(readyEvent);
-                return;
+                RegisterFunctions(_functionRegistrations);
             }
```

Comment:
Can you please add a comment 
Functionsregistrations is null when language worker process is started in placeholder mode

Reply:
Opened https://github.com/Azure/azure-functions-host/issues/3939 to clean this up. Small change but needs more testing. I will address it.